forui/lib/src/foundation/form_field.dart

```diff
@@ -0,0 +1,104 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter/widgets.dart';
+import 'package:meta/meta.dart';
+
+/// A stateless single form field.
+///
+/// This widget is meant to be extended by other form field widgets.
+@internal
+abstract class FFormField<T> extends StatelessWidget {
+  /// An optional method to call with the final value when the form is saved via [FormState.save].
+  final FormFieldSetter<T>? onSave;
+
+  /// An optional property that forces the [FormFieldState] into an error state by directly setting the
+  /// [FormFieldState.errorText] property without running the validator function.
+  ///
+  /// When the [forceErrorText] property is provided, the [FormFieldState.errorText] will be set to the provided value,
+  /// causing the form field to be considered invalid and to display the error message specified.
+  ///
+  /// When [validator] is provided, [forceErrorText] will override any error that it returns. [validator] will not be
+  /// called unless [forceErrorText] is null.
+  final String? forceErrorText;
```

Comment:
Should we make this a widget instead & rename it to `forceError`? It seems a little odd that label and description accept widgets but forceErrorText accepts a string.

Reply:
I used `forceErrorText` because that is the terminology used by Flutter. It needs to accept a string since what's being built is decided by the builder, the form field itself doesn't do any of the building.

https://api.flutter.dev/flutter/widgets/FormField/forceErrorText.html